### PR TITLE
[v14] Add the ability to run a specific tool to Assist.

### PR DESCRIPTION
**Pull request by jakule**
Backport of [#31113](https://github.com/gravitational/teleport/pull/31113)

### CODE REVIEW

**Review by r0mant — APPROVED** [review]

File: lib/ai/client.go [hunk 1]
```diff
@@ -99,6 +99,30 @@ func (client *Client) NewCommand(username string) *Chat {
...
+			&modeltools.AccessRequestListRequestableResourcesTool{})
+	}
+	agent := model.NewAgent(toolContext, tools...)
+	action := &model.AgentAction{
+		Action:    toolName,
+		Input:     toolInput,
+		Reasoning: "Tool invoked directly",
+	}
+
+	return agent.DoAction(ctx, client.svc, action)
```
[the remark this answers is not in the document]
Comment (jakule) [reply lib/ai/client.go:116]: I think we should, but I also don't think we should do that in a backport. We can address those comments in a separate PR targeting master.

File: lib/ai/client_test.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	client := NewClientFromConfig(cfg)
+
+	// Doing the test: Check that the AuditQueryGeneration tool can be invoked
+	// through client.RunTool and validate its response.
+	ctx := context.Background()
+	toolCtx := &tools.ToolContext{User: "alice"}
+	response, _, err := client.RunTool(ctx, toolCtx, tools.AuditQueryGenerationToolName, "List users who connected to a server as root")
+	require.NoError(t, err)
+	message, ok := response.(*output.StreamingMessage)
+	require.True(t, ok)
```
[the remark this answers is not in the document]
Comment (jakule) [reply lib/ai/client_test.go:60]: I think we could change, but I also don't think we should do that in a backport. We can address those comments in a separate PR targeting master.

File: lib/ai/client_test.go [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+	// Doing the test: Check that the EmbeddingRetrieval tool can be invoked
+	// through client.RunTool and validate its response.
+	input := tools.EmbeddingRetrievalToolInput{Question: "Find foobar"}
+	inputText, err := json.Marshal(input)
+	require.NoError(t, err)
+	response, _, err := client.RunTool(ctx, toolCtx, "Nodes names and labels retrieval", string(inputText))
+	require.NoError(t, err)
+	message, ok := response.(*output.Message)
+	require.True(t, ok)
```
[the remark this answers is not in the document]
Comment (jakule) [reply lib/ai/client_test.go:108]: I think we could change, but I also don't think we should do that in a backport. We can address those comments in a separate PR targeting master.

File: lib/ai/model/tools/auditquery.go [hunk 4]
```diff
@@ -29,12 +29,14 @@ import (
 	"github.com/gravitational/teleport/lib/ai/tokens"
 )
 
+const AuditQueryGenerationToolName = "Audit Query Generation"
```
[the remark this answers is not in the document]
Comment (jakule) [reply lib/ai/model/tools/auditquery.go:33]: As above. It can be addressed in a separate PR.

**Review by smallinsky — APPROVED** [review]
LGTM 
Tim left some valid comments but agree with Jakub that we should align it in master first and backport.

**Pull request merged**